File: src/app/model/LapCounterMetric.ts
import { Metric, MetricType } from "./Metric";
import { IMetricService } from "../services/IMetricService";

export class LapCounterMetric extends Metric {
    constructor(metricService: IMetricService) {
        super(MetricType.LAP_COUNTER as MetricType, 'Lap Counter', 'laps', metricService, 'laps', 0);
    }

    startLogging(): void {
        // No external event to subscribe to; lap count is managed internally
    }

    stopLogging(): void {
        // No external event to unsubscribe from
    }

    override newLap(): void {
        super.newLap();
        this.addValue(this.laps.length - 1, new Date());
    }

}

File: android/app/src/main/java/com/openbikecomputer/model/LapCounterMetric.java
package com.openbikecomputer.model;

import com.openbikecomputer.service.IMetricService;

public class LapCounterMetric extends Metric {
    public LapCounterMetric(IMetricService metricService) {
        super(MetricType.LAP_COUNTER, "Lap Counter", "laps", metricService, "laps", 0);
    }

    @Override
    public void startLogging() {
    }

    @Override
    public void stopLogging() {
    }

    @Override
    public void newLap() {
        super.newLap();
        addValue(laps.size() - 1, System.currentTimeMillis());
    }
}
